e2e-tests/gef/cypress/e2e/facility/pre-submission/facility-end-date-enabled/bank-review-date.spec.js

```diff
@@ -0,0 +1,217 @@
+    aboutFacility.shouldCoverStartOnSubmissionYes().click();
+    cy.keyboardInput(aboutFacility.coverEndDateDay(), todayDay);
+    cy.keyboardInput(aboutFacility.coverEndDateMonth(), todayMonth);
+    cy.keyboardInput(aboutFacility.coverEndDateYear(), nextYear);
+    aboutFacility.isUsingFacilityEndDateNo().click();
+
+    cy.clickContinueButton();
+    cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+    cy.fillInBankReviewDate(yesterday);
+
+    cy.clickContinueButton();
+    errorSummary();
+    bankReviewDate.bankReviewDateError();
+
+    cy.fillInBankReviewDate(today);
+
+    cy.clickContinueButton();
+    errorSummary().should('not.exist');
+  });
+
+  it('validates bank review date is less than 6 years in the future', () => {
+    cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+    cy.fillInBankReviewDate(sixYearsOneDay);
+
+    cy.clickContinueButton();
+    errorSummary();
+    bankReviewDate.bankReviewDateError();
+  });
+
+  it('redirects the user to `provided facility` page when form has been successfully filled in', () => {
+    cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+    cy.fillInBankReviewDate(twoYears);
+
+    cy.clickContinueButton();
+
+    cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/provided-facility`));
+  });
+
+  it('stores the inputted values', () => {
+    cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+    cy.fillInBankReviewDate(twoYears);
+
+    cy.clickContinueButton();
+
+    cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+    bankReviewDate.bankReviewDateDay().should('have.value', twoYears.getDate());
+    bankReviewDate.bankReviewDateMonth().should('have.value', twoYears.getMonth() + 1);
+    bankReviewDate.bankReviewDateYear().should('have.value', twoYears.getFullYear());
+  });
+
+  it('redirects to the About Facility page when using facility end date ', () => {
+    cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+    cy.keyboardInput(aboutFacility.facilityName(), 'Name');
+    aboutFacility.shouldCoverStartOnSubmissionYes().click();
+    cy.keyboardInput(aboutFacility.coverEndDateDay(), todayDay);
+    cy.keyboardInput(aboutFacility.coverEndDateMonth(), todayMonth);
+    cy.keyboardInput(aboutFacility.coverEndDateYear(), nextYear);
+    aboutFacility.isUsingFacilityEndDateYes().click();
+    cy.clickContinueButton();
+
+    cy.visit(relative(`/gef/application-details/${application.id}/facilities/${facilityId}/bank-review-date`));
+
+    cy.url().should('eq', relative(`/gef/application-details/${application.id}/facilities/${facilityId}/about-facility`));
+  });
+});
```

Comment:
Same as above, can we look at moving some of the `cy.visit`'s into `beforeEach`'s?